lib/Notification/Notifier.php

```diff
@@ -287,6 +298,75 @@ protected function shortenJsonEncodedMultibyteSave(string $subject, int $dataLen
 		return $temp;
 	}
 
+	private function parseStoredRecording(
+		INotification $notification,
+		Room $room,
+		Participant $participant,
+		IL10N $l
+	): INotification {
+		$shareAction = $notification->createAction()
+			->setParsedLabel($l->t('Share to chat'))
+			->setPrimary(true)
+			->setLink(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'ocs.spreed.Chat.shareObjectToChat',
+					[
+						'apiVersion' => 'v1',
+						'token' => $room->getToken()
+					]
+				),
+				IAction::TYPE_POST
+			);
+		$dismissAction = $notification->createAction()
+			->setParsedLabel($l->t('Dismiss'))
+			->setLink(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'ocs.notifications.Endpoint.deleteNotification',
+					[
+						'apiVersion' => 'v2',
+						'id' => $this->getNotificationId($notification),
+					]
+				),
+				IAction::TYPE_DELETE
+			);
+
+		$notification
+			->setRichSubject(
+				$l->t('Recording for the call in {call} was uploaded.'),
+				[
+					'call' => [
+						'type' => 'call',
+						'id' => $room->getId(),
+						'name' => $room->getDisplayName((string) $participant->getAttendee()->getActorId()),
+						'call-type' => $this->getRoomType($room),
+					],
+				])
+			->addParsedAction($shareAction)
+			->addParsedAction($dismissAction);
+		return $notification;
+	}
+
+	public function getNotificationId(INotification $notification): int {
+		$sql = $this->db->getQueryBuilder();
+		$sql->select('notification_id')
+			->from('notifications')
+			->andWhere($sql->expr()->eq('app',
+				$sql->createNamedParameter($notification->getApp())))
+			->andWhere($sql->expr()->eq('object_id',
+				$sql->createNamedParameter($notification->getObjectId())))
+			->andWhere($sql->expr()->eq('object_type',
+				$sql->createNamedParameter($notification->getObjectType())))
+			->andWhere($sql->expr()->eq('subject',
+				$sql->createNamedParameter($notification->getSubject())))
+			->andWhere($sql->expr()->eq('timestamp',
+				$sql->createNamedParameter($notification->getDateTime()->format('U')),
+				IQueryBuilder::PARAM_INT))
+			->setMaxResults(1);
+
+		$statement = $sql->executeQuery();
+		return (int) $statement->fetchOne();
+	}
```

Comment:
This is not okay, We can not make a query against the notifications tables.
But we can make an endpoint on the recording controller with a method to dismiss the notification and use normal notifications API there by specifying all the fields you did here + the userId and then call `OCP\Notifications\IManager::dismissNotification()`

Reply:
Didn't exist a method to get the notification by other properties and also to dismiss notification.Didn't exist a method to get the notification by other properties and also to dismiss notification.

What do you suggest to get the notification_id or delete a notification using other properties?